resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/localstore/UnifiedTree.java

```diff
@@ -221,11 +228,11 @@ protected void addChildren(UnifiedTreeNode node) {
 			addChildrenMarker();
 	}
 
-	protected void addChildrenFromFileSystem(UnifiedTreeNode node, IFileInfo[] childInfos, int index) {
+	protected void addChildrenFromFileSystem(UnifiedTreeNode node, List<IFileInfo> childInfos, int index) {
 		if (childInfos == null)
 			return;
-		for (int i = index; i < childInfos.length; i++) {
-			IFileInfo info = childInfos[i];
+		for (int i = index; i < childInfos.size(); i++) {
```

Comment:
A counted loop with n calls to size doesn’t look like a nice cleanup.

Reply:
That's right. The 'problem' here is that the loop starts at index and not zero.
But we can instead create a `subList()` here:
`for (IFileInfo info : childInfos.subList(index, childInfos.size())) {`